File: python/raindrops/raindrops.py
def convert(number):
    output = ""
    if number % 3 == 0:
        output = f'{output}Pling'
    if number % 5 == 0:
        output = f'{output}Plang'
    if number % 7 == 0:
        output = f'{output}Plong'
    if output == "":
        output = f'{number}'
    return output

File: ruby/raindrops/raindrops.rb
# Raindrops a bit like fizz buzz
class Raindrops
  SOUND = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }.freeze

  class << self
    def convert(number)
      output = SOUND
               .find_all { |key, _| (number % key).zero? }
               .inject([]) { |ary, (_, value)| ary << value }
      output.count > 0 ? output.join : number.to_s
    end
  end
end

module BookKeeping
  VERSION = 3
end
